packages/datadog-core/src/storage.js

```diff
@@ -2,12 +2,47 @@
 
 const { AsyncLocalStorage } = require('async_hooks')
 
+class DatadogStorage {
+  constructor () {
+    this._storage = new AsyncLocalStorage()
+  }
+
+  disable () {
+    this._storage.disable()
+  }
+
+  enterWith (store) {
+    const handle = {}
```

Comment:
symbol instead ?

Reply:
Sure, but why?

Reply:
Since it's a pretty small change I'll switch to a symbol with the name provided to `storage()`, but I'm still curious as to the benefit in general compared to an object. Not that I disagree with the change, but would still like to know the rationale.

Reply:
I _generally_ prefer symbols too as they can be useful in debugging and other forms of runtime inspection. That said, in _this case_ these objects are really just unique keys for a map without any other meaning, and even mutating them doesn't particularly matter, as they'll keep being compared by identity.

Reply:
Made the change. I think the debugging argument is still valid in this case as it's possible to see the list of stores on the underlying async resource.

Reply:
Turns out this is not supported on Node 18, reverting the change.

Reply:
wow sad